modules/caddyhttp/matchers.go

```diff
@@ -805,20 +810,22 @@ func (m MatchQuery) Match(r *http.Request) bool {
 		// We regard the Go team's decision as sound and thus reject malformed query strings.
 		return false
 	}
-
+	matchedKeys := 0
```

Comment:
I think a code comment is warranted here to clarify the intent

```suggestion

	// Count the amount of matched keys, to ensure we AND
	// between all configured query keys; all keys must
	// match at least one value.
	matchedKeys := 0
```